### PR TITLE
Add to the chai Assert interface

### PR DESCRIPTION
**Pull request by shaunluttin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <http://chaijs.com/api/assert/>
- [x] Increase the version number in the header if appropriate. (Deemed not appropriate).
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by jednano — APPROVED** [review]

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -1659,7 +1659,8 @@ export interface ViewPropertiesAndroid {
 
 type Falsy = undefined | null | false
 interface RecursiveArray<T> extends Array<T | RecursiveArray<T>> {}
-interface RegisteredStyle<T> extends Number {}
+/** Keep a brand of 'T' so that calls to `StyleSheet.flatten` can take `RegisteredStyle<T>` and return `T`. */
+type RegisteredStyle<T> = number & { __registeredStyleBrand: T };
```
Comment (JoshuaKGoldberg) [inline types/react-native/index.d.ts]: Why include this change?

File: types/chai/index.d.ts [hunk 2]
```diff
@@ -488,6 +489,24 @@ declare namespace Chai {
         isNotNaN<T>(value: T, message?: string): void;
 
         /**
+         * Asserts that the target is neither null nor undefined.
+         *
+         * @type T   Type of value.
+         * @param object   Actual value.
```
Comment (JoshuaKGoldberg) [inline types/chai/index.d.ts]: > object

Actual param name is `value`. Here and the other added methods.

**Review by JoshuaKGoldberg — APPROVED** [review]

File: types/react-native/index.d.ts (hunk 1 above)
Comment (shaunluttin) [inline types/react-native/index.d.ts]: Oops.

**Comment by shaunluttin** [comment]
Thank you for the review. I believe I have properly address the comments.

**Pull request merged**